app/src/docs/v1.api-spec.yaml

```diff
@@ -0,0 +1,2592 @@
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  valid:
+                    type: boolean
+                    description: Whether the Activity ID is valid and exists
+                    example: true
+        '400':
+          description: Invalid Activity ID format
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Invalid activity Id format
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          $ref: '#/components/responses/Error'
+  /document/:
+    put:
+      summary: Create or update a document
+      description: >-
+        Creates a link between an activity and a previously existing object in
+        COMS.
+      operationId: createDocument
+      tags:
+        - Document
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Request-Document'
+      responses:
+        '201':
+          description: Document created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Document'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        default:
+          $ref: '#/components/responses/Error'
+  /document/{documentId}:
+    delete:
+      summary: Delete a document by ID
+      description: Deletes the document associated with the provided document ID.
+      operationId: deleteDocument
+      tags:
+        - Document
+      parameters:
+        - $ref: '#/components/parameters/Path-DocumentId'
+      responses:
+        '200':
+          description: Document deleted successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DB-Document'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        default:
+          $ref: '#/components/responses/Error'
+  /document/list/{activityId}:
+    get:
+      summary: List documents by Activity ID
+      description: Retrieves a list of documents associated with a given activity.
+      operationId: listDocuments
+      tags:
+        - Document
+      parameters:
+        - $ref: '#/components/parameters/Path-ActivityId'
+      responses:
+        '200':
+          description: A list of documents for the specified activity
+          content:
+            application/json:
+              schema:
```

Comment:
return object is slightly different createdAt and createdBy is also being returned.
`[
    {
        "documentId": "aaeecdfd-79cd-4a7c-a772-d906637bcb63",
        "activityId": "F2172B78",
        "filename": "dsadsadsa_4c080e3c_ec07a13f.docx",
        "mimeType": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
        "filesize": 11855,
        "createdAt": "2024-08-30T16:00:21.199Z",
        "createdBy": "7255f440-1f4a-431b-9548-b7ce2e9e6fb4",
        "createdByFullName": "Babu, Sanjay WLRS:EX"
    }
]`